sync/sync.go

```diff
@@ -57,15 +58,23 @@ func (s *Synchronizer) fetcherTask(ctx context.Context, height uint64, verifiers
 			if err != nil {
 				continue
 			}
+			declaredClasses := make(map[felt.Felt]*core.Class, len(stateUpdate.StateDiff.DeployedContracts))
+			for _, classHash := range stateUpdate.StateDiff.DeclaredClasses {
+				class, err := s.StarknetData.Class(ctx, classHash)
```

Comment:
Could we make DeclaredClasses in StateUpdate a map and move this to gatewayclient?

Reply:
I would love to do that, but it is a very invasive change. 

Since only the class hashes (not the full classes themselves) are present in the state update response from the gateway, we would need to send separate `get_class` requests for each hash. We could do that in two places:

1. Within the state update adapt function (not preferable: limits the caller's ability to parallelize requests and the behavior goes beyond the what the function name indicates).
2. In the `sync` package and pass the received classes into the `starknetdata.Gateway.StateUpdate` function. This leaks the gateway implementation to the `starknetdata` interface, which is what we're trying to avoid.

Reply:
> Within the state update adapt function (not preferable: limits the caller's ability to parallelize requests and the behavior goes beyond the what the function name indicates).

I would actually prefer this. Let's see what others think.

Reply:
That's fair. @IronGauntlets what do you think?

Reply:
I prefer `1.` as well. 

I don't think the behaviour goes beyond what the function indicates. All we are doing is getting all the declared contracts (classes), which are part of the state update. That was what I did in the previous implementation. [Here](https://github.com/NethermindEth/juno/blob/jelilat/implement-class-storage/starknetdata/gateway/gateway.go#L334)

Reply:
I would also prefer `1`. However, we don't need to do it in this PR.

Reply:
Thanks everyone for the feedback. I think I'll handle this in another PR.